src/btree/bt_random.c

```diff
@@ -408,13 +409,15 @@ __wt_random_descent(WT_SESSION_IMPL *session, WT_REF **refp, uint32_t flags, WT_
         descent = NULL;
         for (i = 0; i < entries; ++i) {
             descent = pindex->index[__wt_random(rnd) % entries];
-            if (descent->state == WT_REF_DISK || descent->state == WT_REF_MEM)
+            descent_state = __wt_atomic_loadv8(&descent->state);
```

Comment:
I assume these are locations where WT_READ_ONCE is more appropriate?

Reply:
Yep these are the candidates. Making the change just depends on our taste for functional changes in what is meant to be more of a refactor PR

Reply:
I would say do it, if we notice it its better to do it than leave it.

Reply:
Updated in [6a18404](https://github.com/wiredtiger/wiredtiger/pull/10328/commits/6a18404e99790a24e809af814dfcf4fed89d4be2)
There's once case where I couldn't use use `READ_ONCE` due to a a 32bit typecast (`slot_last_offset` in `__wt_log_wrlsn`). I'll come back to this case in [WT-12610](https://jira.mongodb.org/browse/WT-12610) when we review the src/log/* files.